samples/subsys/fs/littlefs/src/main.c

```diff
@@ -12,11 +13,249 @@
 #include <device.h>
 #include <fs/fs.h>
 #include <fs/littlefs.h>
+#include <logging/log.h>
 #include <storage/flash_map.h>
 
+LOG_MODULE_REGISTER(main);
+
 /* Matches LFS_NAME_MAX */
 #define MAX_PATH_LEN 255
+#define TEST_FILE_SIZE 547
+
+static uint8_t file_test_pattern[TEST_FILE_SIZE];
+static int lsdir(const char *path)
+{
+	int res;
+	struct fs_dir_t dirp;
+	static struct fs_dirent entry;
+
+	fs_dir_t_init(&dirp);
+
+	/* Verify fs_opendir() */
+	res = fs_opendir(&dirp, path);
+	if (res) {
+		LOG_ERR("Error opening dir %s [%d]\n", path, res);
+		return res;
+	}
+
+	LOG_PRINTK("\nListing dir %s ...\n", path);
+	for (;;) {
+		/* Verify fs_readdir() */
+		res = fs_readdir(&dirp, &entry);
+
+		/* entry.name[0] == 0 means end-of-dir */
+		if (res || entry.name[0] == 0) {
+			break;
+		}
```

Comment:
There is no log on `res != 0` here, so there is no difference between empty dir and error, as you do in other cases. If you rely on `fs_readdir` logging error here, this is fine just make sure VFS logs are on.

Reply:
I've added some extra `LOG_ERR()` function to present the error.